config/v1/types_scheduling.go

```diff
@@ -48,6 +51,14 @@ type SchedulerSpec struct {
 	// would not be applied.
 	// +optional
 	DefaultNodeSelector string `json:"defaultNodeSelector,omitempty"`
+	// MaxMastersSchedulable allows a certain percentage of available master nodes to
+	// be schedulable. As of now, the supported values are 0 and 100. In future, we 
+	// will allow a certain percentage of master nodes to be schedulable. For example,
+	// if we have 3 running master nodes in the cluster, setting this value to 33% means, 1
+	// node of 3 available master nodes would become schedulable. The absolute value is calculated
+	// by rounding down. This field is optional
+	// +optional
+	MaxMastersSchedulable *intstr.IntOrString `json:"percentageOfMastersSchedulable"`
```

Comment:
I think this should be a bool.  I don't see a reason to make this a percentage.

Reply:
The reason is in future, say if we have 5 masters cluster. We can tell, I want only 2 to be schedulable, so that even if they're loaded and they go down, we can still maintain quorum(with 3 masters). The other use-case is developers telling, I want 2 nodes to be made worker nodes and let one run as master..

Reply:
@soltysh had a concern about making this a `bool`, specifically having to change this interface later if we ever wanted to to allow the user to specify a subset of masters that should be scheduleable.

@deads2k thoughts on that?

Reply:
> @soltysh had a concern about making this a `bool`, specifically having to change this interface later if we ever wanted to to allow the user to specify a subset of masters that should be scheduleable.
> 
> @deads2k thoughts on that?

If that becomes a thing we want (and I think that's pretty unlikely), we can add another field that takes precedence over this one.